var/spack/repos/builtin/packages/py-mpi4py/package.py

```diff
@@ -17,6 +17,7 @@ class PyMpi4py(PythonPackage):
     git = "https://github.com/mpi4py/mpi4py.git"
 
     version("master", branch="master")
+    version("3.1.3", sha256="f1e9fae1079f43eafdd9f817cdb3fd30d709edc093b5d5dada57a461b2db3008")
     version("3.1.2", sha256="40dd546bece8f63e1131c3ceaa7c18f8e8e93191a762cd446a8cfcf7f9cce770")
     version("3.1.1", sha256="e11f8587a3b93bb24c8526addec664b586b965d83c0882b884c14dc3fd6b9f5c")
     version("3.1.0", sha256="134fa2b2fe6d8f91bcfcc2824cfd74b55ca3dcbff4d185b1bda009beea9232ec")
```

Comment:
`py-setuptools` should be `@40.9:`, at least in newer releases, didn't go all the way back. See `pyproject.toml` for where this version constraint is imposed.

Reply:
Are you suggesting:
```diff
-    depends_on("py-setuptools", type="build")
+    depends_on("py-setuptools@40.9:", type="build")
```
or someother change?


Also cc: @dalcinl

Reply:
Yes. If you want to be even more thorough, you could find the first version that includes that in `pyproject.toml` and add a `when` clause, but I'd be fine with the above change too.

Reply:
Thanks! pushed the above change

Reply:
@adamjstewart I'm not sure what `depends_on("py-setuptools@40.9:", type="build")` actually means, but let me clarify a few things:

The mpi4py version constraint in `pyproject.toml` is `setuptools>=40.9`, not `setuptools==40.9`. Nonetheless, an older setuptools version may still work (although `pyproject.toml` may be ignored). In fact, I believe that mpi4py can still build with no setuptools at all.

In current mpi4py master, I'm using `setuptools>=42`. If you look at setuptools [changelog](https://setuptools.pypa.io/en/latest/history.html#v42-0-0), v42 added support for `python_requires`, and that's something mpi4py's latest release is using in `setup.py`, so it may be the case that using `setuptools==40.9` could cause trouble with the parsing of `python_requires`.

Reply:
> I'm not sure what `depends_on("py-setuptools@40.9:", type="build")` actually means

`40.9:` (with a trailing colon) means `>=40.9`

> In fact, I believe that mpi4py can still build with no setuptools at all.

All Python libraries need some kind of build backend in order to install. `pyproject.toml` says to use setuptools as the build backend. Although the `setup.py` first tries setuptools and then falls back on distutils, frontend build tools like `pip` will actually replace distutils with setuptools automatically.